backend/app/api/v1/commons/ocp.py

```diff
@@ -4,26 +4,38 @@
 from app.services.search import ElasticService
 
 
-async def getData(start_datetime: date, end_datetime: date, configpath: str):
+async def getData(
+    start_datetime: date,
+    end_datetime: date,
+    size: int,
+    offset: int,
+    sort: str,
+    configpath: str,
+):
     query = {
+        "size": size,
+        "from": offset,
         "query": {
             "bool": {"filter": {"range": {"timestamp": {"format": "yyyy-MM-dd"}}}}
-        }
+        },
     }
+    if sort:
+        key, direction = sort.split(":")
+        query["sort"] = [{key: {"order": direction}}]
```

Comment:
You added field names and direction constants to the service; but here you're just splitting and accepting the result without any validation.

FYI, I built a method for this in my Crucible service...

(a bit more complex since I allowed multiple sort terms...)

```python
    @classmethod
    def _build_sort_terms(cls, sorters: Optional[list[str]]) -> list[dict[str, str]]:
        """Build sort term list

        Sorters may reference any native `run` index field and must specify
        either "asc"(ending) or "desc"(ending) sort order. Any number of
        sorters may be combined, like ["name:asc,benchmark:desc", "end:desc"]

        Args:
            sorters: list of <key>:<direction> sort terms

        Returns:
            list of OpenSearch sort terms
        """
        if sorters:
            sort_terms = []
            for s in sorters:
                key, dir = s.split(":", maxsplit=1)
                if dir not in cls.DIRECTIONS:
                    raise HTTPException(
                        status.HTTP_400_BAD_REQUEST,
                        f"Sort direction {dir!r} must be one of {','.join(cls.DIRECTIONS)}",
                    )
                if key not in cls.FIELDS:
                    raise HTTPException(
                        status.HTTP_400_BAD_REQUEST,
                        f"Sort key {key!r} must be one of {','.join(cls.FIELDS)}",
                    )
                sort_terms.append({f"run.{key}": {"order": dir}})
        else:
            sort_terms = [{"run.begin": {"order": "asc"}}]
        return sort_terms
```

Reply:
Patternfly table can support sorting on only one column at a time. So, I didn't go for multiple sort terms.